src/main/java/com/fakng/fakngagrgtr/FakngAgrgtrApplication.java

```diff
@@ -1,14 +1,32 @@
 package com.fakng.fakngagrgtr;
 
+import com.fakng.fakngagrgtr.parser.apple.AppleParser;
+import com.fakng.fakngagrgtr.persistent.vacancy.VacancyRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.boot.CommandLineRunner;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.scheduling.annotation.EnableScheduling;
 
+import java.util.concurrent.CompletableFuture;
+
 @SpringBootApplication
 @EnableScheduling
-public class FakngAgrgtrApplication {
+@RequiredArgsConstructor
+public class FakngAgrgtrApplication implements CommandLineRunner {
 
+    private final AppleParser appleParser;
+    private final VacancyRepository vacancyRepository;
     public static void main(String[] args) {
         SpringApplication.run(FakngAgrgtrApplication.class, args);
     }
+
+    @Override
+    public void run(String... args) throws Exception {
+        CompletableFuture
+                .supplyAsync(appleParser::parse)
+                .thenApply(vacancyRepository::saveAll)
+                .thenCompose(vacancies -> CompletableFuture.supplyAsync(() -> appleParser.parseSecondary(vacancies)))
```

Comment:
So when primary parsing is done, vacancies are saved in DB in an incomplete state. The description is not fully constructed. We can even prevent displaying of such entities using the new flag I added. But then there's the secondary async step, which is about collecting more granular data for each vacancy from its own page. When that's done, all entities are updated in bulk once again.